app/models/ems_refresh/save_inventory.rb

```diff
@@ -219,7 +220,10 @@ def save_disks_inventory(hardware, hashes)
       h[:storage_profile_id] = h.fetch_path(:storage_profile, :id)
     end
 
-    save_inventory_multi(hardware.disks, hashes, :use_association, [:controller_type, :location], nil, [:storage, :backing, :storage_profile])
+    filenames = hashes.collect { |h| h[:filename] }
+
+    save_inventory_mulit_child(:disks, Disk.where(:filename => filenames), parent, hashes,
+                               [:filename, :device_name], nil, [:storage, :backing, :storage_profile])
```

Comment:
I know that filename and device_name is rhev specific but I spend a lot of time to understand that location won't work due to [index](https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/redhat/infra_manager/refresh_parser.rb#L541)